tox.ini

```diff
@@ -139,10 +139,11 @@ commands = codespell --ignore-words=.config/codespell_ignore.txt --skip="*.pyc,*
 description = "Check Scapy code distribution"
 skip_install = true
 deps = twine
-       setuptools>=38.6.0
        cmarkgfm
-commands = python setup.py --quiet sdist
-           twine check dist/*
+       build
+setenv = SCAPY_VERSION=3.0.0
+commands = python -m build
+           twine check --strict dist/*
```

Comment:
I think it would be great to eventually add something like
```sh
git archive HEAD >scapy.tar
python3 -m pip install scapy.tar # or just unpack it and run `python3 -m build` inside the directory instead
```
just to make sure that tarballs are buildable and installable. It can be done later though.

Reply:
No you're right, let's do this now to prevent regressions.

Reply:
Thanks for adding the test! It passes locally.

As far as I can tell it failed on the CI because the "checkout" action cloned it with `--depth=1`:
```sh
git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +417a1525419b9db10b379aedb221cefc277c5fbc:refs/remotes/pull/3958/merge
git checkout remotes/pull/3958/merge
```
and `git_archive_id` was `'1681043547 '` instead of `'1681043522 v2.5.0-72-gedb31f4e'`. It can probably be fixed by using https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches